sql-plugin/src/main/scala/org/apache/spark/sql/rapids/stringFunctions.scala

```diff
@@ -1379,3 +1379,107 @@ case class GpuStringSplit(str: Expression, regex: Expression, limit: Expression)
       limit: GpuScalar): ColumnVector =
     throw new IllegalStateException("This is not supported yet")
 }
+
+class GpuStringToMapMeta(
+    expr: StringToMap,
+    conf: RapidsConf,
+    parent: Option[RapidsMeta[_, _, _]],
+    rule: DataFromReplacementRule)
+    extends TernaryExprMeta[StringToMap](expr, conf, parent, rule) {
+  import GpuOverrides._
+
+  private def checkRegex(delimExpr: Expression) : Unit = {
+    extractLit(delimExpr).foreach { delim =>
+      val str = delim.value.asInstanceOf[UTF8String]
+      if (str != null){
+        //
+        // Currently, we don't support regex. However, we can't check for regex very accurately.
```

Comment:
I personally would check for the default string delims and add them to an allow list. That way it will still work by default and we can at least check it in with everything turned on, then in a follow on issue we can add in support for regular expressions.